File: Dopamine.SettingsModule/ViewModels/FolderViewModel.vb
Imports Microsoft.Practices.Prism.Mvvm
Imports Dopamine.Core.Database
Imports Dopamine.Core.Database.Entities

Public Class FolderViewModel
    Inherits BindableBase

#Region "Variables"
    Private mFolder As Folder
#End Region

#Region "Properties"
    Public Property Folder() As Folder
        Get
            Return mFolder
        End Get
        Set(ByVal value As Folder)
            SetProperty(Of Folder)(Me.mFolder, value)
            OnPropertyChanged(Function() Me.Path)
            OnPropertyChanged(Function() Me.Directory)
        End Set
    End Property

    Public ReadOnly Property Path() As String
        Get
            Return Me.Folder.Path
        End Get
    End Property

    Public ReadOnly Property Directory() As String
        Get
            Return System.IO.Path.GetFileName(Me.Folder.Path)
        End Get
    End Property

    Public Property ShowInCollection() As Boolean
        Get
            Return If(Me.Folder.ShowInCollection = 1, True, False)
        End Get
        Set(ByVal value As Boolean)

            If value Then
                Me.Folder.ShowInCollection = 1
            Else
                Me.Folder.ShowInCollection = 0
            End If

            OnPropertyChanged(Function() Me.ShowInCollection)
        End Set
    End Property
#End Region
End Class

File: Dopamine.SettingsModule/ViewModels/FolderViewModel.cs
using Dopamine.Common.Database.Entities;
using Prism.Mvvm;

namespace Dopamine.SettingsModule.ViewModels
{
    public class FolderViewModel : BindableBase
    {
        #region Variables
        private Folder folder;
        #endregion

        #region Properties
        public Folder Folder
        {
            get { return this.folder; }
            set
            {
                SetProperty<Folder>(ref this.folder, value);
                OnPropertyChanged(() => this.Path);
                OnPropertyChanged(() => this.Directory);
            }
        }

        public string Path
        {
            get { return this.Folder.Path; }
        }

        public string Directory
        {
            get { return System.IO.Path.GetFileName(this.Folder.Path); }
        }

        public bool ShowInCollection
        {
            get { return this.Folder.ShowInCollection == 1 ? true : false; }

            set
            {
                if (value)
                {
                    this.Folder.ShowInCollection = 1;
                }
                else
                {
                    this.Folder.ShowInCollection = 0;
                }

                OnPropertyChanged(() => this.ShowInCollection);
            }
        }
        #endregion
    }
}
